lib/src/chart/line_chart/line_chart_painter.dart

```diff
@@ -326,4 +333,85 @@ class LineChartPainter extends AxisChartPainter {
 
   @override
   bool shouldRepaint(CustomPainter oldDelegate) => false;
+
+  void drawAnnotation(Canvas canvas, Size viewSize) {
+    if (data.extraLinesData != null && data.extraLinesData.show) {
+      if (data.extraLinesData.horizontalLines != null) {
+        for (HorizontalLine line in data.extraLinesData.horizontalLines) {
+          _drawHorizontalLineLine(canvas, viewSize, line);
+        };
+      }
+      if (data.extraLinesData.verticalLines != null) {
+        for (VerticalLine line in data.extraLinesData.verticalLines) {
+          _drawVerticalLineLine(canvas, viewSize, line);
+        }
+      }
+      return;
+    }
+  }
+
+  void _drawHorizontalLineLine(Canvas canvas, Size viewSize, HorizontalLine line) {
+    viewSize = getChartUsableDrawSize(viewSize);
+    extraLinesPaint.color = line.color;
+    extraLinesPaint.strokeWidth = line.strokeWidth;
+    final double x = line.endX == null ? viewSize.width : getPixelX(line.endX, viewSize);
+    final double y = getPixelY(line.y, viewSize);
+    if (line.dashDefinition != null) {
+      _drawDashedLine(canvas, extraLinesPaint, 0, y, x, y, line.dashDefinition);
+    } else {
+      _drawSolidLine(canvas, extraLinesPaint, 0, y, x, y);
```

Comment:
remove this if else and determine the line as a solid line every time ( I will add Line Style after your PR in a separate phase), and you can draw the solid line here and you don't need to put it in a new function (because there is no if else anymore)

Reply:
Removed